src/Form/CloudFlareAdminSettingsForm.php

```diff
@@ -2,20 +2,100 @@
 
 /**
  * @file
- * Contains Drupal\cloudflare\Form\DefaultForm.
+ * Contains Drupal\cloudflare\Form\CloudFlareAdminSettingsForm.
  */
 
 namespace Drupal\cloudflare\Form;
 
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
 use Drupal\Core\Form\ConfigFormBase;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\cloudflare\CloudFlareStateInterface;
+use Drupal\cloudflare\CloudFlareZoneInterface;
+use Egulias\EmailValidator\EmailValidator;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use CloudFlarePhpSdk\Exceptions\CloudFlareException;
 
 /**
- * Class DefaultForm.
+ * Class CloudFlareAdminSettingsForm.
  *
  * @package Drupal\cloudflare\Form
  */
-class CloudFlareAdminSettingsForm extends ConfigFormBase {
+class CloudFlareAdminSettingsForm extends ConfigFormBase implements ContainerInjectionInterface {
```

Comment:
Thinking out loud about the UX for both our modules. What about abstracting this form out into a Trait to reuse its code in this form **and** in a offered  ``\Drupal\purge_ui\Form\PurgerConfigFormBase`` derivative? That way users can configure the API in both places, without duplicating logic and risking bugz.